pype/plugins/global/publish/extract_review.py

```diff
@@ -880,28 +892,61 @@ def rescaling_filters(self, temp_data, output_def, new_repre):
 
         # letter_box
         if letter_box:
+            letter_box_ratio = letter_box
+            if isinstance(letter_box, dict):
+                letter_box_ratio = letter_box["ratio"]
+
             if input_res_ratio == output_res_ratio:
-                letter_box /= pixel_aspect
+                letter_box_ratio /= pixel_aspect
             elif input_res_ratio < output_res_ratio:
-                letter_box /= scale_factor_by_width
+                letter_box_ratio /= scale_factor_by_width
             else:
-                letter_box /= scale_factor_by_height
+                letter_box_ratio /= scale_factor_by_height
 
-            scale_filter = "scale={}x{}:flags=lanczos".format(
-                output_width, output_height
-            )
+            filters.extend([
+                "scale={}x{}:flags=lanczos".format(
+                    output_width, output_height
+                ),
+                "setsar=1"
+            ])
 
-            top_box = (
-                "drawbox=0:0:iw:round((ih-(iw*(1/{})))/2):t=fill:c=black"
-            ).format(letter_box)
+        if letter_box and isinstance(letter_box, float):
+            filters.extend(self.get_letterbox_filters(letter_box))
 
-            bottom_box = (
-                "drawbox=0:ih-round((ih-(iw*(1/{0})))/2)"
-                ":iw:round((ih-(iw*(1/{0})))/2):t=fill:c=black"
-            ).format(letter_box)
+        if letter_box and isinstance(letter_box, dict):
```

Comment:
Why it's not part of `get_letterbox_filters`?

Reply:
Probably a bad name actually for that method. Its to reuse the code for getting the boxes for the letterbox mode. Open to suggetions.

Reply:
If it would contain whole logic (with current lines 854 - 949) the name would make sence.

Reply:
How about calling it `get_top_bottom_box_filters`?

Reply:
And why not put the whole logic to the function instead? It's more than 100 lines of code related to "only this".

Reply:
Its because that part of the code is used twice. Once for backward compatibility with `letterbox` being a float and once for `letterbox` being a dictionary.

Reply:
```
def get_letterbox_filters(self, letter_box_def):
    if not letter_box_def:
        return []
    
    if not isinstance(letter_box_def, dict):
        ratio = letter_box_def["ratio"]
        state = letter_box_def["state"]
        thickness = letter_box_def["thickness"]
        color = letter_box_def["color"]
    else:
        ratio = letter_box_def
        state = "letterbox"
        thickness = "fill"
        color = "black"

    if state == "letterbox":
        ...
    elif state == "pillar":
        ...
    else:
        ...
```

Reply:
Cool, done.